[BUG][BE1] data race

### Description (Actual behavior)
Data race encountered during a run of the tests

<details>
<summary>Trace</summary>
<pre>
=== RUN   TestConnectToSocket
2022-06-14T19:02:21+02:00 INF mod.go:151 > connected to organizer at ws://localhost:9001/organizer/witness
2022-06-14T19:02:21+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="organizer socket" to=127.0.0.1:9001
2022-06-14T19:02:21+02:00 INF ../network/socket/socket.go:75 > listening for messages from organizer role="organizer socket"
2022-06-14T19:02:21+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="organizer socket" to=127.0.0.1:9001
2022-06-14T19:02:21+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="organizer socket"
2022-06-14T19:02:21+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="organizer socket"
--- PASS: TestConnectToSocket (4.00s)
=== RUN   TestOrganizerAndWitness
    pop_test.go:20: running Organizer: [/tmp/go-build3606310257/b204/cli.test organizer --pk J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= serve --witness-port 9011 --client-port 8010]
{"level":"info","time":"2022-06-14T19:02:21+02:00","message":"The owner public key has been specified, only J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= can create LAO"}
2022-06-14T19:02:23+02:00 INF ../network/server.go:75 > setting handler at /organizer/client for port 8010 role=server
2022-06-14T19:02:23+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8010 role=server
2022-06-14T19:02:23+02:00 INF ../network/server.go:75 > setting handler at /organizer/witness for port 9011 role=server
2022-06-14T19:02:23+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9011 role=server
    pop_test.go:30: organizer up
    pop_test.go:36: running Witness 1: [/tmp/go-build3606310257/b204/cli.test witness --pk J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= serve --witness-port 9003 --organizer-address localhost:9011 --client-port 8020]
{"level":"info","time":"2022-06-14T19:02:23+02:00","message":"The owner public key has been specified, only J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= can create LAO"}
2022-06-14T19:02:25+02:00 INF ../network/server.go:75 > setting handler at /witness/client for port 8020 role=server
2022-06-14T19:02:25+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8020 role=server
2022-06-14T19:02:25+02:00 INF ../network/server.go:75 > setting handler at /witness/witness for port 9003 role=server
2022-06-14T19:02:25+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9003 role=server
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:49484
2022-06-14T19:02:25+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:49484 requestID=1655226145336318671 role=server url=/organizer/witness
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:25+02:00 INF mod.go:151 > connected to organizer at ws://localhost:9011/organizer/witness
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:75 > listening for messages from organizer role="organizer socket"
2022-06-14T19:02:25+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:25+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:25+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:25+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="witness socket" to=127.0.0.1:49484
2022-06-14T19:02:25+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:25+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:25+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:25+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:25+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
    pop_test.go:46: witness 1 up
    pop_test.go:52: running Witness 2: [/tmp/go-build3606310257/b204/cli.test witness --pk J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= serve --witness-port 9004 --organizer-address localhost:9011 --client-port 8021 --other-witness localhost:9003]
{"level":"info","time":"2022-06-14T19:02:25+02:00","message":"The owner public key has been specified, only J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= can create LAO"}
2022-06-14T19:02:27+02:00 INF ../network/server.go:75 > setting handler at /witness/client for port 8021 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8021 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:75 > setting handler at /witness/witness for port 9004 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9004 role=server
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:49498
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:49498 requestID=1655226147505386898 role=server url=/organizer/witness
2022-06-14T19:02:27+02:00 INF mod.go:151 > connected to organizer at ws://localhost:9011/organizer/witness
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:75 > listening for messages from organizer role="organizer socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="witness socket" to=127.0.0.1:49498
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:55588
2022-06-14T19:02:27+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:55588 requestID=1655226147540495465 role=server url=/witness/witness
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:27+02:00 INF mod.go:151 > connected to witness at ws://localhost:9003/witness/witness
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="witness socket" to=127.0.0.1:9003
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:9003
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="witness socket" to=127.0.0.1:55588
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:27+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
    pop_test.go:62: witness 2 up
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8010 role=server
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8021 role=server
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8020 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9011 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9004 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="organizer socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="organizer socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="organizer socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
    pop_test.go:58: witness 2 done
2022-06-14T19:02:27+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9003 role=server
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
    pop_test.go:26: organizer done
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
2022-06-14T19:02:27+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
2022-06-14T19:02:27+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="organizer socket"
    pop_test.go:42: witness 1 done
--- PASS: TestOrganizerAndWitness (6.02s)
=== RUN   TestOrganizerAndWitnessWithoutPK
    pop_test.go:75: running Organizer: [/tmp/go-build3606310257/b204/cli.test organizer serve --witness-port 9011 --client-port 8010]
{"level":"info","time":"2022-06-14T19:02:27+02:00","message":"No public key specified for the owner, everyone can create LAO."}
2022-06-14T19:02:27+02:00 INF ../network/server.go:93 > stopped the server... role=server
    pop_test.go:85: organizer up
==================
WARNING: DATA RACE
Write at 0x00c0001d3f50 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x726
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3f50 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c0001d3f60 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x81e
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3f60 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c0001d3f70 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x86b
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3f70 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c0001d3f80 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x8b8
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3f80 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c0001d3f90 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x905
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3f90 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c0001d3fa0 by goroutine 40:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:89 +0x952
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0001d3fa0 by goroutine 41:
  [failed to restore the stack]

Goroutine 40 (running) created at:
  testing.(*T).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:47 +0x22b

Goroutine 41 (running) created at:
  popstellar/cli.TestOrganizerAndWitnessWithoutPK()
      /home/popdev/popstellar/be1-go/cli/pop_test.go:77 +0x628
  testing.tRunner()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /nix/store/vwkz5x8x1k82s6y7br5i5lf2x9adjrr9-go-1.17.10/share/go/src/testing/testing.go:1306 +0x47
==================
    pop_test.go:91: running Witness 1: [/tmp/go-build3606310257/b204/cli.test witness serve --witness-port 9003 --organizer-address localhost:9011 --client-port 8020]
{"level":"info","time":"2022-06-14T19:02:29+02:00","message":"No public key specified for the owner, everyone can create LAO."}
2022-06-14T19:02:30+02:00 INF ../network/server.go:75 > setting handler at /organizer/client for port 8010 role=server
2022-06-14T19:02:30+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8010 role=server
2022-06-14T19:02:30+02:00 INF ../network/server.go:75 > setting handler at /organizer/witness for port 9011 role=server
2022-06-14T19:02:30+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9011 role=server
2022-06-14T19:02:31+02:00 INF ../network/server.go:75 > setting handler at /witness/client for port 8020 role=server
2022-06-14T19:02:31+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8020 role=server
2022-06-14T19:02:31+02:00 INF ../network/server.go:75 > setting handler at /witness/witness for port 9003 role=server
2022-06-14T19:02:31+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9003 role=server
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:50018
2022-06-14T19:02:31+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:50018 requestID=1655226151633953322 role=server url=/organizer/witness
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:31+02:00 INF mod.go:151 > connected to organizer at ws://localhost:9011/organizer/witness
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:75 > listening for messages from organizer role="organizer socket"
2022-06-14T19:02:31+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:31+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:31+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:31+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:31+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:31+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:31+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="witness socket" to=127.0.0.1:50018
2022-06-14T19:02:31+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:31+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
    pop_test.go:101: witness 1 up
    pop_test.go:107: running Witness 2: [/tmp/go-build3606310257/b204/cli.test witness serve --witness-port 9004 --organizer-address localhost:9011 --client-port 8021 --other-witness localhost:9003]
{"level":"info","time":"2022-06-14T19:02:31+02:00","message":"No public key specified for the owner, everyone can create LAO."}
2022-06-14T19:02:33+02:00 INF ../network/server.go:75 > setting handler at /witness/client for port 8021 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:85 > starting to listen at: localhost:8021 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:75 > setting handler at /witness/witness for port 9004 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:85 > starting to listen at: localhost:9004 role=server
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:50034
2022-06-14T19:02:33+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:50034 requestID=1655226153512158077 role=server url=/organizer/witness
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:33+02:00 INF mod.go:151 > connected to organizer at ws://localhost:9011/organizer/witness
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:75 > listening for messages from organizer role="organizer socket"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="organizer socket" to=127.0.0.1:9011
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":0,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="witness socket" to=127.0.0.1:50034
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":0,\"result\":[]}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:54754
2022-06-14T19:02:33+02:00 INF ../network/server.go:171 >  agent=Go-http-client/1.1 method=GET remoteAddr=127.0.0.1:54754 requestID=1655226153532539535 role=server url=/witness/witness
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:33+02:00 INF mod.go:151 > connected to witness at ws://localhost:9003/witness/witness
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:75 > listening for messages from witness role="witness socket"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="witness socket" to=127.0.0.1:9003
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:166 > send generic msg msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="witness socket" to=127.0.0.1:9003
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"method\":\"catchup\",\"id\":1,\"params\":{\"channel\":\"/root\"}}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:238 > send result msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="witness socket" to=127.0.0.1:54754
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:455 > handle incoming message msg="{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[]}" role="base hub"
2022-06-14T19:02:33+02:00 INF ../validation/schema_validator.go:59 > verifying msg follows the schema role="base hub"
    pop_test.go:117: witness 2 up
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8010 role=server
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8021 role=server
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:8020 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9011 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9003 role=server
2022-06-14T19:02:33+02:00 INF ../network/shutdown.go:22 > shutting down server localhost:9004 role=server
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../network/server.go:93 > stopped the server... role=server
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:215 > waiting for existing workers to finish... role="base hub"
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="organizer socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="organizer socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:154 > closing the write pump role="witness socket"
    pop_test.go:81: organizer done
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="organizer socket"
    pop_test.go:97: witness 1 done
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="organizer socket"
2022-06-14T19:02:33+02:00 INF ../network/socket/socket.go:92 > closing the read pump role="witness socket"
    pop_test.go:113: witness 2 done
    testing.go:1152: race detected during execution of test
--- FAIL: TestOrganizerAndWitnessWithoutPK (6.03s)
2022-06-14T19:02:33+02:00 INF ../hub/standard_hub/mod.go:205 > stopping the hub role="base hub"
=== CONT  
    testing.go:1152: race detected during execution of test
FAIL
FAIL	popstellar/cli	16.376s
</pre>
</details>

### Expected behavior
No data race

### How to reproduce
1. Run `make check`
2. Get lucky

### Version & Environment
This bug was reproduced on:
- Commit: 63fd4bee083a417f0399ac5bb812f59149fbe38e

##### Front-ends:
- [ ] Fe1-Web (please include browser's names & version)
- [ ] Fe2-Android (please specify if phone or emulation, and Android version)
- [x] Not applicable

##### Back-ends:
- [x] Be1-Go
- [ ] Be2-Scala
- [ ] Not applicable

##### Environment (as applicable):
 - Device: DH67CL desktop computer
 - OS: `Linux 5.15.43, NixOS, 22.05 (Quokka), 22.05.20220612.1f8d880`
 - Browser: N/A

### Impact
Could cause safety violations in production.

### Possible root cause
There's a race condition somewhere.

--- a/be1-go/cli/pop_test.go
+++ b/be1-go/cli/pop_test.go
@@ -11,16 +11,16 @@ import (
 const waitUp = time.Second * 2
 
 func TestOrganizerAndWitness(t *testing.T) {
-	args := os.Args[0:1]
-	args = append(args, "organizer", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9011", "--client-port", "8010")
 
 	ctx, cancel := context.WithCancel(context.Background())
 	wait := sync.WaitGroup{}
 
-	t.Logf("running Organizer: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "organizer", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9011", "--client-port", "8010"}
+		t.Logf("running Organizer: %v", args)
 
 		run(ctx, args)
 		t.Log("organizer done")
@@ -30,13 +30,13 @@ func TestOrganizerAndWitness(t *testing.T) {
 	t.Log("organizer up")
 
 	// Test Witness -> Organizer
-	args = args[0:1]
-	args = append(args, "witness", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9003", "--organizer-address", "localhost:9011", "--client-port", "8020")
 
-	t.Logf("running Witness 1: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "witness", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9003", "--organizer-address", "localhost:9011", "--client-port", "8020"}
+		t.Logf("running Witness 1: %v", args)
 
 		run(ctx, args)
 		t.Log("witness 1 done")
@@ -46,13 +46,13 @@ func TestOrganizerAndWitness(t *testing.T) {
 	t.Log("witness 1 up")
 
 	// Test Witness -> Witness
-	args = args[0:1]
-	args = append(args, "witness", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9004", "--organizer-address", "localhost:9011", "--client-port", "8021", "--other-witness", "localhost:9003")
 
-	t.Logf("running Witness 2: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "witness", "--pk", "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=", "serve", "--witness-port", "9004", "--organizer-address", "localhost:9011", "--client-port", "8021", "--other-witness", "localhost:9003"}
+		t.Logf("running Witness 2: %v", args)
 
 		run(ctx, args)
 		t.Log("witness 2 done")
@@ -66,16 +66,16 @@ func TestOrganizerAndWitness(t *testing.T) {
 }
 
 func TestOrganizerAndWitnessWithoutPK(t *testing.T) {
-	args := os.Args[0:1]
-	args = append(args, "organizer", "serve", "--witness-port", "9011", "--client-port", "8010")
 
 	ctx, cancel := context.WithCancel(context.Background())
 	wait := sync.WaitGroup{}
 
-	t.Logf("running Organizer: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "organizer", "serve", "--witness-port", "9011", "--client-port", "8010"}
+		t.Logf("running Organizer: %v", args)
 
 		run(ctx, args)
 		t.Log("organizer done")
@@ -85,13 +85,13 @@ func TestOrganizerAndWitnessWithoutPK(t *testing.T) {
 	t.Log("organizer up")
 
 	// Test Witness -> Organizer
-	args = args[0:1]
-	args = append(args, "witness", "serve", "--witness-port", "9003", "--organizer-address", "localhost:9011", "--client-port", "8020")
 
-	t.Logf("running Witness 1: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "witness", "serve", "--witness-port", "9003", "--organizer-address", "localhost:9011", "--client-port", "8020"}
+		t.Logf("running Witness 1: %v", args)
 
 		run(ctx, args)
 		t.Log("witness 1 done")
@@ -101,13 +101,13 @@ func TestOrganizerAndWitnessWithoutPK(t *testing.T) {
 	t.Log("witness 1 up")
 
 	// Test Witness -> Witness
-	args = args[0:1]
-	args = append(args, "witness", "serve", "--witness-port", "9004", "--organizer-address", "localhost:9011", "--client-port", "8021", "--other-witness", "localhost:9003")
 
-	t.Logf("running Witness 2: %v", args)
 	wait.Add(1)
 	go func() {
 		defer wait.Done()
+
+		args := []string{os.Args[0], "witness", "serve", "--witness-port", "9004", "--organizer-address", "localhost:9011", "--client-port", "8021", "--other-witness", "localhost:9003"}
+		t.Logf("running Witness 2: %v", args)
 
 		run(ctx, args)
 		t.Log("witness 2 done")

